treeherder/webapp/api/runnable_jobs.py

```diff
@@ -79,7 +79,7 @@ def list(self, request, project):
 
         for label, node in tc_graph.iteritems():
             task_metadata = node['task']['metadata']
-            treeherder_options = node['task']['extra']['treeherder']
+            treeherder_options = node['task'].get('extra', {}).get('treeherder', {})
```

Comment:
@jgraham we would need to change all fields which use `treeherder_options` in the subsequent code. Given the uncertainties, I guess it's best to use `get()` everywhere :(